src/Corrade/Utility/Test/DirectoryTest.cpp

```diff
@@ -933,9 +937,22 @@ void DirectoryTest::list() {
         "CTest is not able to run XCTest executables properly in the simulator.");
     #endif
 
+    #ifdef CORRADE_TARGET_WINDOWS
+    DWORD handleCount = 0;
+    GetProcessHandleCount(GetCurrentProcess(), &handleCount);
+    #endif
+
     CORRADE_COMPARE_AS(Directory::list(_testDir),
         (std::vector<std::string>{".", "..", "dir", "file"}),
         TestSuite::Compare::SortedContainer);
+
+    #ifdef CORRADE_TARGET_WINDOWS
+    /* Ensure we are not leaking handles */
+    DWORD newHandleCount = 0;
+    GetProcessHandleCount(GetCurrentProcess(), &newHandleCount);
```

Comment:
Wow, this is cool, didn't even know such thing was possible.

You know what? This would be good to do for everything -- I think this could be doable in setup/teardown routines, but I need to patch TestSuite a bit to allow checks there. Is it okay if I merge just the first part with the scope guard so you have the fix and leave this extra check for later when TestSuite is ready?

Reply:
We can also merge this and refactor it once test suite is ready (or refactor with making it ready).